File: 0434-20-2965-find-missing-and-repeated-values/0434-20-2965-find-missing-and-repeated-values.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        int a = 0;
        int gsum = 0;
        for (int[] row : grid) {
            for (int x : row) {
                int nx = Math.abs(x);
                gsum += nx;
                int r = (nx - 1) / n;
                int c = (nx - 1) % n;
                if (grid[r][c] < 0) {
                    a = nx;
                }
                grid[r][c] *= -1;
            }
        }
        int nSquared = n * n;
        int totalSum = nSquared * (nSquared + 1) / 2;
        int[] result = new int[2];
        result[0] = a;
        result[1] = totalSum - gsum + a;
        return result;
    }
}

File: 0434-20-2965-find-missing-and-repeated-values/0434-20-2965-find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid):
        n = len(grid)
        a = 0
        gsum = 0
        for row in grid:
            for x in row:
                nx = abs(x)
                gsum += nx
                r = (nx - 1) // n
                c = (nx - 1) % n
                if grid[r][c] < 0:
                    a = nx
                grid[r][c] *= -1
        n_squared = n * n
        total_sum = n_squared * (n_squared + 1) // 2
        return [a, total_sum - gsum + a]